contracts/pallet-ibc/src/client.rs

```diff
@@ -118,55 +118,45 @@ where
 		client_id: &ClientId,
 		height: Height,
 	) -> Result<Option<AnyConsensusState>, ICS02Error> {
-		let mut cs_states = ConsensusStates::<T>::iter_key_prefix(client_id)
-			.map(|(height, cs_state)| {
-				let cs = AnyConsensusState::decode_vec(&cs_state).map_err(|e| {
-					ICS02Error::implementation_specific(format!(
-						"[next_consensus_state]: error decoding consensus state from bytes {}",
-						e
-					))
-				})?;
-				Ok((height, cs))
+		let consensus_heights = ConsensusHeights::<T>::get(client_id.as_bytes().to_vec());
```

Comment:
do we need to operate in a `Vac`? Can't we stay on the slice level and avoid an allocation?

Reply:
The get method does not accept a reference